includes/widgets/class-wc-widget-rating-filter.php

```diff
@@ -98,28 +98,29 @@ public function widget( $args, $instance ) {
 		ob_start();
 
 		$found         = false;
-		$rating_filter = isset( $_GET['rating_filter'] ) ? array_filter( array_map( 'absint', explode( ',', wp_unslash( $_GET['rating_filter'] ) ) ) ) : array(); // WPCS: input var ok, CSRF ok, sanitization ok.
+		$rating_filter = isset( $_GET['rating_filter'] ) ? absint( wp_unslash( $_GET['rating_filter'] ) ) : ''; // WPCS: input var ok, CSRF ok, sanitization ok.
 
 		$this->widget_start( $args, $instance );
 
 		echo '<ul>';
 
+		$link = get_permalink( wc_get_page_id( 'shop' ) );
+		if ( is_product_category() ) {
+			$link = get_term_link( get_queried_object_id() );
+		}
```

Comment:
The link logic doesn't take into account product tags and other filters that may be active (e.g. "Sort by popularity").

Reply:
Good point. Working on it.

Reply:
Fixed this issue. Please have a look. :)